### DISCUSSION TITLE
tuya smart thermostat integration

### DISCUSSION THREAD

**Discussion by Mikeep** [ScratMan/HASmartThermostat#107] [Q&A]
![image](https://user-images.githubusercontent.com/32244645/214258800-fdaab4cd-d429-4bcf-a758-5ccaa8777701.png)

hi i have a tuya smart thermostat integrated with home assistant all sensors showing 

the problem i have is that when i set a temp on the screen of the above image it does not adjust the set_temp in ha i believe because this set in the line target_temp it has a numeric value usually set to 19 but when adjusted in ha dashboard this changes , how do i get this to set at the same as the sensor set_temp from the thermostat as that line will not accept anything other than a numerical value 

how this makes sense 

many thanks mike

**Comment by ScratMan** [comment]
You may try with an automation in HA if the temperature setting of this thermostat exposes an entity available in HA. The automation should be triggered by this entity's state change or a state change of the virtual thermostat in HA, conditioned by the thermostats being on, then the action should take the new temperature from the thermostat that changed, and apply it to the other one.
